From 66fa708894f7f59c95be70c77eb657bc9e1bd6a9 Mon Sep 17 00:00:00 2001
From: Pan Junzhong <junzhong.pan@spacemit.com>
Date: Thu, 22 Aug 2024 10:10:47 +0800
Subject: [PATCH 0964/1206] phy-k1x-ci-otg: fix shared reset assert warning

Current reset cause warning becasue deassert after assert
cannot apply to shared without deasserted first. Fix it by
making reset assert and deassert symmetry.

Fix this warning:
[    1.579546] WARNING: CPU: 4 PID: 1 at drivers/reset/core.c:457 reset_control_assert+0xe6/0x122
[    1.579568] Modules linked in:
[    1.579577] CPU: 4 PID: 1 Comm: swapper/0 Not tainted 6.6.36+ #2
[    1.579583] Hardware name: M1-MUSE-PAPER (DT)
[    1.579586] epc : reset_control_assert+0xe6/0x122
[    1.579591]  ra : mv_otg_enable_internal+0x44/0xdc
........
[    1.579638] status: 0000000200000120 badaddr: 0000000000000000 cause: 0000000000000003
[    1.579642] [<ffffffff808664ca>] reset_control_assert+0xe6/0x122
[    1.579648] [<ffffffff80abcbe2>] mv_otg_enable_internal+0x44/0xdc

Change-Id: I01ae83364f5345fb35e56e18d8f0ece111e5aa1d
---
 drivers/usb/phy/phy-k1x-ci-otg.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/usb/phy/phy-k1x-ci-otg.c b/drivers/usb/phy/phy-k1x-ci-otg.c
index 4bd7ebfd093c..522135bef288 100644
--- a/drivers/usb/phy/phy-k1x-ci-otg.c
+++ b/drivers/usb/phy/phy-k1x-ci-otg.c
@@ -229,7 +229,6 @@ static int mv_otg_enable_internal(struct mv_otg *mvotg)
 		"otg enabled, will enable clk, release rst\n");
 
 	otg_clock_enable(mvotg);
-	otg_reset_assert(mvotg);
 	otg_reset_deassert(mvotg);
 	retval = usb_phy_init(mvotg->outer_phy);
 	if (retval) {
-- 
2.47.0

